### DISCUSSION TITLE
Enforcing determinism with drjit

### DISCUSSION THREAD

**Discussion by naperman** [NVlabs/sionna#953] [Q&A]
I've adopted the [calibrating material parameters through gradient descent](calibrating-material-parameters-through-gradient-descent) developer guide example for a simple outdoor scene with very few objects(2 buildings and ground plane), but I'm not able to get convergence with ground truth material properties. There is a decent number of transmitters(UL) sampled on the radio map to ensure sufficient interaction with objects in the scene(around 1000) and sufficient number of samples per source(300K). I'm wondering if it comes down to some non-determinism in the path solver when it is repeatedly called within the gradient descent loop. In tensorflow, it is possible to enforce determinism using [tf.config.experimental.enable_op_determinism()
](https://www.tensorflow.org/api_docs/python/tf/config/experimental/enable_op_determinism) - I've observed that this step needs to performed even after settings the seed with sionna APIs. Is there a similar way to enforce determinism with drjit?

**Comment by faycalaa** [comment]
Hi @naperman ,

Please have a look at [this thread](https://github.com/NVlabs/sionna/discussions/851) regarding the determinism of Sionna RT. However, randomness in path tracing should not be an issue for calibration.

Concerning the calibration of radio materials, it is generally a problem with many local minima. Therefore, calibrating the radio materials through the channel impulse responses (CIRs) could lead to convergence to material properties different from the ground-truth ones, but resulting in very close or identical CIRs. Could that be what you observe?
